src/Particles/babylon.solidParticleSystem.ts

```diff
@@ -0,0 +1,427 @@
+        private _shapeCounter = 0;
+        private _useParticleColor = true;
+        private _useParticleTexture = true;
+        private _useParticleRotation = true;
+        private _useParticleVertex = false;
+        private _cam_axisZ = Vector3.Zero();
+        private _cam_axisY = Vector3.Zero();
+        private _cam_axisX = Vector3.Zero();
+        private _axisX = Axis.X;
+        private _axisY = Axis.Y;
+        private _axisZ = Axis.Z;
+        private _camera: Camera;
+        private _particle: SolidParticle;
+        private _fakeCamPos = Vector3.Zero();
+        private _rotMatrix = new Matrix();
+        private _invertedMatrix = new Matrix();
+        private _rotated = Vector3.Zero();
+        private _quaternion = new Quaternion();
+        private _vertex = Vector3.Zero();
+        private _yaw = 0.0;
+        private _pitch = 0.0;
+        private _roll = 0.0;
+        private _halfroll = 0.0;
+        private _halfpitch = 0.0;
+        private _halfyaw = 0.0;
+        private _sinRoll = 0.0;
+        private _cosRoll = 0.0;
+        private _sinPitch = 0.0;
+        private _cosPitch = 0.0;
+        private _sinYaw = 0.0;
+        private _cosYaw = 0.0;
+
+        constructor(name: string, scene: Scene) {
+            this.name = name;
+            this._scene = scene;
+            this._camera = scene.activeCamera;
+        }
+
+        // build the SPS mesh : returns the mesh
+        public buildMesh(): Mesh {
+            if (this.nbParticles == 0) {
+                //this.addTriangles(1, 1);
+            }
+            VertexData.ComputeNormals(this._positions, this._indices, this._normals);
+            var vertexData = new VertexData();
+            vertexData.positions = this._positions;
+            vertexData.indices = this._indices;
+            vertexData.normals = this._normals;
+            if (this._uvs) {
+                vertexData.uvs = this._uvs;
+            }
+            if (this._colors) {
+                vertexData.colors = this._colors;
+            }
+            var mesh = new Mesh(name, this._scene);
+            vertexData.applyToMesh(mesh, true);
+            this.mesh = mesh;
+            return mesh;
+        }
+
+
+        // adds a new particle object in the particles array
+        private _addParticle(p: number, idxpos: number, shape: Vector3[], shapeUV: number[], shapeId: number): void {
+            this.particles.push(new SolidParticle(p, idxpos, shape, shapeUV, shapeId));
+        }
+
+        // add solid particles from a shape model in the particles array
+        public addShape(mesh: Mesh, nb: number): number {
+            var meshPos = mesh.getVerticesData(VertexBuffer.PositionKind);
+            var meshInd = mesh.getIndices();
+            var meshUV = mesh.getVerticesData(VertexBuffer.UVKind);
+            var meshCol = mesh.getVerticesData(VertexBuffer.ColorKind);
+            // shape and shapeUV
+            var posToShape = (positions) => {
```

Comment:
ok, this will be done

Le 14/10/2015 17:21, David Catuhe a écrit :
>
> In src/Particles/babylon.solidParticleSystem.ts 
> <https://github.com/BabylonJS/Babylon.js/pull/751#discussion_r42007871>:
>
> > +        }
> > +
> > +
> > +        // adds a new particle object in the particles array
> > +        private _addParticle(p: number, idxpos: number, shape: Vector3[], shapeUV: number[], shapeId: number): void {
> > +            this.particles.push(new SolidParticle(p, idxpos, shape, shapeUV, shapeId));
> > +        }
> > +
> > +        // add solid particles from a shape model in the particles array
> > +        public addShape(mesh: Mesh, nb: number): number {
> > +            var meshPos = mesh.getVerticesData(VertexBuffer.PositionKind);
> > +            var meshInd = mesh.getIndices();
> > +            var meshUV = mesh.getVerticesData(VertexBuffer.UVKind);
> > +            var meshCol = mesh.getVerticesData(VertexBuffer.ColorKind);
> > +            // shape and shapeUV
> > +            var posToShape = (positions) => {
>
> You should not create this function each time. Just extract it and 
> mark it as private
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/BabylonJS/Babylon.js/pull/751/files#r42007871>.
>